oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStoreTest.java

```diff
@@ -0,0 +1,710 @@
+                                final DataRecord rhs,
+                                boolean lastModifiedEquals)
+            throws DataStoreException, IOException {
+        validateRecord(record, contents, rhs.getIdentifier(), rhs.getLength(), rhs.getLastModified(), lastModifiedEquals);
+    }
+
+    private void validateRecord(final DataRecord record,
+                                final String contents,
+                                final DataIdentifier identifier,
+                                final long length,
+                                final long lastModified)
+            throws DataStoreException, IOException {
+        validateRecord(record, contents, identifier, length, lastModified, true);
+    }
+
+    private void validateRecord(final DataRecord record,
+                                final String contents,
+                                final DataIdentifier identifier,
+                                final long length,
+                                final long lastModified,
+                                final boolean lastModifiedEquals)
+            throws DataStoreException, IOException {
+        assertEquals(record.getLength(), length);
+        if (lastModifiedEquals) {
+            assertEquals(record.getLastModified(), lastModified);
+        } else {
+            assertTrue(record.getLastModified() > lastModified);
+        }
+        assertTrue(record.getIdentifier().toString().equals(identifier.toString()));
+        StringWriter writer = new StringWriter();
+        org.apache.commons.io.IOUtils.copy(record.getStream(), writer, "utf-8");
+        assertTrue(writer.toString().equals(contents));
+    }
+
+    private static InputStream randomStream(int seed, int size) {
+        Random r = new Random(seed);
+        byte[] data = new byte[size];
+        r.nextBytes(data);
+        return new ByteArrayInputStream(data);
+    }
+
+    private static String getIdForInputStream(final InputStream in)
+            throws NoSuchAlgorithmException, IOException {
+        MessageDigest digest = MessageDigest.getInstance("SHA-1");
+        OutputStream output = new DigestOutputStream(new NullOutputStream(), digest);
+        try {
+            IOUtils.copyLarge(in, output);
+        } finally {
+            IOUtils.closeQuietly(output);
+            IOUtils.closeQuietly(in);
+        }
+        return encodeHexString(digest.digest());
+    }
+
+    private static int countIteratorEntries(final Iterator<?> iter) {
+        int ctr = 0;
+        while (iter.hasNext()) { iter.next(); ++ctr; }
+        return ctr;
+    }
+
+    @Test
+    public void testCreateAndDeleteBlobHappyPath() throws DataStoreException, IOException {
+        final DataRecord uploadedRecord = ds.addRecord(new ByteArrayInputStream(testBuffer));
+        DataIdentifier identifier = uploadedRecord.getIdentifier();
+        //DataIdentifier identifier = new DataIdentifier("a94a8fe5ccb19ba61c4c0873d391e987982fbbd3");
```

Comment:
This comment was kept in my initial prototype for local testing purposes.  It should be deleted now.